dhis-2/dhis-services/dhis-service-tracker/src/main/java/org/hisp/dhis/tracker/validation/hooks/EventDateValidationHook.java

```diff
@@ -64,14 +64,21 @@ public void validateEvent( ValidationErrorReporter reporter, Event event )
     {
         TrackerImportValidationContext context = reporter.getValidationContext();
 
-        if ( event.getOccurredAt() == null )
+        Program program = context.getProgram( event.getProgram() );
+
+        if ( event.getOccurredAt() == null && occuredAtDateIsMandatory( event, program ) )
         {
             reporter.addError( newReport( TrackerErrorCode.E1031 )
                 .addArg( event ) );
             return;
         }
 
-        Program program = context.getProgram( event.getProgram() );
+        if ( event.getScheduledAt() == null && EventStatus.SCHEDULE == event.getStatus() )
+        {
+            reporter.addError( newReport( TrackerErrorCode.E1050 )
```

Comment:
Consider using the "shortcut" version here `addError(reporter, E1050, event)`

Reply:
Yes, done that.